cdisc-org/cdisc-rules-engine
# Iss 315 move cdisc library conformance rules generator to GitHub

Moves ADO cdisc-library-conformance-rules-generator repo to cdisc-rules-engine in GH.

Made the following transformations to the generator:
- Deleted unused endpoint test files to fix broken unit tests
- Moved function app files to root
- Converted pipeline to workflow
- Merged unit tests
- Imported relative cdisc-rules-engine instead of using requirements package
- Linted
- updated hosts extension bundle version
- remove enums and use enums from cdisc-rules-engine

Made following changes to cdisc-rules-engine:
- Updated automated-CI linter versions to match pre-commit linter versions
- Created secret AZURE_SP (service principal)
- Merged gitignores
- Merged requirements

To test you can double check:
- Workflow has been tested and run successfully. View the `Deploy Azure Rule Tester Endpoint` Actions. To rerun yourself:
  `gh workflow run "Deploy Azure Rule Tester Endpoint" --ref ISS-315-move-cdisc-library-conformance-rules-generator-to-github`
- Confirm that the function app code has updated
- Confirm that rule-editor can still access the execute rule endpoint successfully
